sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,130 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+/// Represents ltree specific errors
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum Error {
+    /// LTree labels can only contain [A-Za-z0-9_]
+    #[error("ltree label cotains invalid characters")]
+    InvalidLtreeLabel,
+
+    /// LTree version not supported
+    #[error("ltree version not supported")]
+    InvalidLtreeVersion,
+}
+
+/// Represents an postgres ltree. Not that this is an EXTENSION!
```

Comment:
```suggestion
/// Container for a Label Tree (`ltree`) in Postgres.
///
/// See https://www.postgresql.org/docs/current/ltree.html
///
/// ### Note: Extension Required
/// The `ltree` extension is not enabled by default in Postgres. You will need to do so explicitly:
///
/// ```ignore
/// CREATE EXTENSION IF NOT EXISTS "ltree";
/// ```
```